subsys/net/lib/nrf_cloud/Kconfig

```diff
@@ -66,6 +76,22 @@ config NRF_CLOUD_NONBLOCKING_SEND
 	  into such a situation, at the cost of failing in situations where
 	  a short period of blocking would have sent the data.
 
+config NRF_CLOUD_SEND_TIMEOUT
+	bool "Send data with socket timeout"
+	help
+	  Configures a timeout on the nRF Cloud socket to ensure that a call to
+	  te send function will not block indefinitely.
+	  To configure the length of the timeout, use NRF_CLOUD_SEND_TIMEOUT_SEC.
+
+endchoice
+
+config NRF_CLOUD_SEND_TIMEOUT_SEC
+	int "Send timeout"
+	default 60
```

Comment:
`depends on NRF_CLOUD_SEND_TIMEOUT`?

Reply:
Yeah, I had that initially, but then the use in `nrf_cloud_transport.c` gives a compilaton error as `CONFIG_NRF_CLOUD_SEND_TIMEOUT_SEC` is undefined if `CONFIG_NRF_CLOUD_SEND_TIMEOUT` is not used.
Couldn't see another way to solve it than introduce some ifdeffing or leave it like this

Reply:
Understood. I guess it's still better to keep using `IS_ENABLED(CONFIG_NRF_CLOUD_SEND_TIMEOUT)` instead of ifdef so indeed there might be no other way around.

Reply:
That's what I thought, too. I'll leave it as is then.